enhancements/network/routed-ingress-primary-udn-static-ips.md

```diff
@@ -0,0 +1,723 @@
+deploying an application in a cluster.
+
+1. The cluster creator sits down at their keyboard...
+2. ...
+3. The cluster creator sees that their cluster is ready to receive
+   applications, and gives the application administrator their
+   credentials.
+
+See
+https://github.com/openshift/enhancements/blob/master/enhancements/workload-partitioning/management-workload-partitioning.md#high-level-end-to-end-workflow
+and https://github.com/openshift/enhancements/blob/master/enhancements/agent-installer/automated-workflow-for-agent-based-installer.md for more detailed examples.
+
+### API Extensions
+
+#### IPAMClaim CRD
+
+The IPAMClaim CRD status sub-resource will need to be updated, adding
+conditions.
+
+For traceability, we also suggest adding to the `IPAMClaim` spec an attribute
+to indicate which pod is holding the claim at any given time. On a VM live
+migration, the OVN-Kubernetes control plane would update the `OwnerPod` after
+the claim has been consumed by a different pod. Same for VM start/stop
+scenarios.
+
+If we choose the [de-centralized IP management](#de-centralized-ip-management)
+alternative, we will also need to change the IPAMClaim CRD spec, adding it an
+attribute to request specific IPs for the workload which will consume the
+claim.
+
+Below you'll see the changes requested:
+```go
+type IPAMClaimSpec struct {
+	// The network name for which this persistent allocation was created
+	Network string `json:"network"`
+	// The pod interface name for which this allocation was created
+	Interface string `json:"interface"`
++	// The IPs requested by the user
++	// +optional
++	IPRequests []CIDR `json:"ipRequests,omitempty"`
+}
+
+// IPAMClaimStatus contains the observed status of the IPAMClaim.
+type IPAMClaimStatus struct {
+    // The list of IP addresses (v4, v6) that were allocated for the pod interface
+-   IPs []string `json:"ips"`
++   IPs []CIDR `json:"ips"`
++   // The name of the pod holding the IPAMClaim
++   OwnerPod string `json:"ownerPod"`
++   // Conditions contains details for one aspect of the current state of this API Resource
++   Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+```
+
+The `IPAMClaim` status will have (at least) the following conditions:
+- SuccessfulAllocation: reports the IP address was successfully allocated for
+  the workload
+- AllocationConflict: reports the requested allocation was not successful - i.e.
+  the requested IP address is already present in the network
+
+#### New IPPool CRD
+
+The IPPool CRD will operate as a place to store the MAC to IP addresses
+association for a logical network.
+
+```go
+type IPPool struct {
+    metav1.TypeMeta     `json:",inline"`
+    metav1.ObjectMeta   `json:"metadata,omitempty"`
+
+	Spec   IPPoolSpec   `json:"spec,omitempty"`
+	Status IPPoolStatus `json:"status,omitempty"`
+}
+
+type IPPoolSpec struct {
+    NetworkName string                      `json:"network-name"`
+	Entries map[net.HardwareAddr][]net.IP   `json:"entries"`
```

Comment:
Do we need this to be a CIDR ? 
```suggestion
	Entries map[net.HardwareAddr][]net.IPNet   `json:"entries"`
```
The NSE ip request is a cidr
```yaml
annotations:
    k8s.v1.cni.cncf.io/networks: '[
            {
                "name": "macvlan1-config",
                "ips": [ "10.1.1.11/24" ],
                "interface": "net1"
            }
    ]'
```

Maybe is dangerous to do so.